src/Tests/CaptchaCacheTestCase.php

```diff
@@ -0,0 +1,96 @@
+<?php
+
+/**
+ * @file
+ * Contains Drupal\captcha\Tests\CaptchaCacheTestCase.
+ */
+
+namespace Drupal\captcha\Tests;
+
+use Drupal\captcha\Entity\CaptchaPoint;
+use Drupal\Core\Url;
+
+/**
+ * Tests CAPTCHA caching on various pages.
+ *
+ * @group captcha
+ */
+class CaptchaCacheTestCase extends CaptchaBaseWebTestCase {
+
+  public static $modules = ['block', 'image_captcha'];
+
+  function setUp() {
+    parent::setUp();
+
+    $this->adminUser = $this->drupalCreateUser(array('administer blocks'));
+    $this->drupalLogin($this->adminUser);
+    $this->drupalPlaceBlock('user_login_block', array('id' => 'login'));
+    $this->drupalLogout($this->adminUser);
+  }
+
+  /**
+   * Test the cache tags.
+   */
+  public function testCacheTags() {
+    // Check caching without captcha as anonymous user.
+    $this->drupalGet('');
+    $this->assertEqual($this->drupalGetHeader('x-drupal-cache'), 'MISS');
+    $this->drupalGet('');
+    $this->assertEqual($this->drupalGetHeader('x-drupal-cache'), 'HIT');
+
+    // Repeat the same after enabling captcha/Math.
+    captcha_set_form_id_setting('user_login_form', 'captcha/Math');
+    $this->drupalGet('');
+    $sid = $this->getCaptchaSidFromForm();
+    $math_challenge = (string) $this->xpath('//span[@class="field-prefix"]')[0];
+    $this->assertFalse($this->drupalGetHeader('x-drupal-cache'), 'Cache is disabled');
+    $this->drupalGet('');
+    $this->assertNotEqual($sid, $this->getCaptchaSidFromForm());
+    $this->assertNotEqual($math_challenge, (string) $this->xpath('//span[@class="field-prefix"]')[0]);
+
+    // Repeat the same after setting the captcha challange to captcha/Test.
+    captcha_set_form_id_setting('user_login_form', 'captcha/Test');
+    $this->drupalGet('');
+    $sid = $this->getCaptchaSidFromForm();
+    $this->assertFalse($this->drupalGetHeader('x-drupal-cache'), 'Cache is disabled');
+    $this->drupalGet('');
+    $this->assertNotEqual($sid, $this->getCaptchaSidFromForm());
+
+    // Repeat the same after setting the captcha challange to captcha/Image.
+    captcha_set_form_id_setting('user_login_form', 'image_captcha/Image');
+    $this->drupalGet('');
+    $image_path = (string) $this->xpath('//img[2]//@src')[0];
+    $this->assertFalse($this->drupalGetHeader('x-drupal-cache'), 'Cache disabled');
+    // Check that we get a new image when vising the page again.
+    $this->drupalGet('');
+    $this->assertNotEqual($image_path, (string) $this->xpath('//img[2]//@src')[0]);
+    // Check image caching.
```

Comment:
sthe comments below don't make too much sense. Say something like "Request the image twice twice to make sure that no errors happen (e.g. due to page caching)" and remove the comments below.